cookbook/email/gmail.rst

```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
             'password'  => 'your_gmail_password',
         ));
 
-You're done!
+If you are using the Symfony Standard Edition, it's more convenient to configure
+these parameters in ``parameters.yml``:
 
-.. tip::
+.. code-block:: yaml
 
-    If you are using the Symfony Standard Edition, configure the parameters in ``parameters.yml``:
+    # app/config/parameters.yml
+    parameters:
+        # ...
+        mailer_user:      your_gmail_username
+        mailer_password:  your_gmail_password
+
+.. configuration-block::
 
     .. code-block:: yaml
 
-        # app/config/parameters.yml
-        parameters:
-            # ...
-            mailer_transport: gmail
-            mailer_host:      ~
-            mailer_user:      your_gmail_username
-            mailer_password:  your_gmail_password
+        # app/config/config_dev.yml
+        swiftmailer:
+            transport: gmail
+            username:  %mailer_user%
+            password:  %mailer_password%
+
+    .. code-block:: xml
+
+        <!-- app/config/config_dev.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:swiftmailer="http://symfony.com/schema/dic/swiftmailer"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/swiftmailer
+                http://symfony.com/schema/dic/swiftmailer/swiftmailer-1.0.xsd">
+
+            <!-- ... -->
+            <swiftmailer:config
+                transport="gmail"
+                username="%mailer_user%"
+                password="%mailer_password%"
+            />
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config_dev.php
+        $container->loadFromExtension('swiftmailer', array(
+            'transport' => 'gmail',
+            'username'  => '%mailer_user%',
+            'password'  => '%mailer_password%',
+        ));
+
+Redefining the Default Configuration Parameters
+-----------------------------------------------
 
-.. note::
+The ``gmail`` transport is simply a shortcut that uses the ``smtp`` transport
+and sets ``encryption`` to ``ssl``, ``auth_mode`` to ``login`` and ``host`` to
+``smtp.gmail.com`` to work with Gmail.
 
-    The ``gmail`` transport is simply a shortcut that uses the ``smtp`` transport
-    and sets ``encryption``, ``auth_mode`` and ``host`` to work with Gmail.
+If your application uses ``tls`` encryption or ``oauth`` authentication, you
+must override the default options by defining the ``encryption`` and ``auth_mode``
+parameters.
 
-.. note::
+If your Gmail account uses 2-Step-Verification, you must `generate an App password`_
+to use for your ``mailer_password`` parameter. You must also ensure that you
```

Comment:
~~~
If your Gmail account uses 2-Step-Verification, you must `generate an App password`_
to use for your ``mailer_password`` parameter.
~~~

sounds weird, what about:
~~~
If you are using 2-Step-Verification, you must `generate an App password`_ and use this as your ``mailer_password``.
~~~